clang/lib/StaticAnalyzer/Checkers/StdLibraryFunctionsChecker.cpp

```diff
@@ -1427,8 +1427,13 @@ void StdLibraryFunctionsChecker::checkPostCall(const CallEvent &Call,
             });
         Pred = C.addTransition(NewState, Pred, Tag);
       }
-      if (!Pred)
+      if (!Pred || Pred->isSink()) {
+        // Pred may be:
+        //  - a nullpointer, when the generated node is not new;
+        //  - a sink, when NewState is posteriorly overconstrained.
+        // In these situations we cannot add the errno note tag.
```

Comment:
I like more if the comment is added before the `if` and the braces are not added.

Reply:
I updated the patch.